addon/components/validated-input/component.js

```diff
@@ -29,6 +29,16 @@ export default Ember.Component.extend({
         defineProperty(this, 'value', computed.alias(`model.${valuePath}`));
     },
 
+    // This is a to make sure we parse and set the date value from manual inputs.
+    focusOut: function () {
+        if (this.get('isDateField')) {
+            let date = Date.parse(this.$('input')[0].value);
+            if (!isNaN(date)) {
+                this.set('value', new Date(date));
+            }
+        }
+    },
```

Comment:
This validation should be done with [ember-cp-validations](https://github.com/offirgolan/ember-cp-validations) instead